alternator/executor.cc

```diff
@@ -347,6 +351,24 @@ get_table_or_view(service::storage_proxy& proxy, const rjson::value& request) {
     }
 }
 
+// get_table_for_write() is similar to get_table(), but additionally, if the
+// configuration allows this, may also allow writing to system table with
+// prefix INTERNAL_TABLE_PREFIX. This is analogous to the function
+// get_table_or_view() above which allows *reading* internal tables.
+static schema_ptr get_table_for_write(service::storage_proxy& proxy, const rjson::value& request) {
```

Comment:
I think it may be a bit unclear to future programmer when to use get_table_for_write / get_table / get_table_or_view

ideally there should be a single function, access control shouldn't anyway depend on finding the table

 I suspect it's done because some functions won't support system tables, maybe a bool argument like `include_system` would be a solution

Reply:
We already we have get_table vs get_table_or_view for the obvious reason - when you read you can read from the base table or GSI or LSI so you need get_table_or_view, but when you write you can only write to the base table, not a view.

But now we need another split - the get_table, the one used for the *write* (and can't find the view) now has to be split to one which is really used for write (and can also write to system tables), and various table operations (like DescribeTable) which can only operate on real tables. This is why I ended up with three similar functions.

The "with_system" is an obscure implementation detail (and in fact, in get_table_for_write() is even optional), so I thought a name describing the use of the function (i.e., get_table_for_write() is used in writes) will be clearer, and added comments mentioning the system tables thing. I'll try to see if I can improve it further.

Reply:
ok